packages/ui/src/components/editor/properties/light/shadowProperties/index.tsx

```diff
@@ -45,9 +52,35 @@ type LightShadowPropertiesProps = {
  */
 export const LightShadowProperties: EditorComponentType = (props: LightShadowPropertiesProps) => {
   const { t } = useTranslation()
-
+  const shadowMapResolution = getState(RendererState).shadowMapResolution
+  const cameraEntity = getState(ReferenceSpaceState).viewerEntity
+  const camera = getComponent(cameraEntity, CameraComponent)
+  const { scene } = getComponent(getState(ReferenceSpaceState).viewerEntity, RendererComponent)
   const lightComponent = useComponent(props.entity, props.component) as any
 
+  const calculateShadowBias = () => {
+    const boundingBox = new Box3()
+
+    scene.traverse((object) => {
```

Comment:
we need to use entity functions (iterateEntityNode) instead of threejs scene traversal